### PR TITLE
Rename `get_safe_execution_payload_hash` -> `*_block_hash`

### PR DESCRIPTION
**Pull request by etan-status**
The name `get_safe_execution_payload_hash` is misleading, as it returns the execution block hash. There is no concept of a payload hash. Rename the function, and also update some documentation for clarity.

### CODE REVIEW

**Review by mkalinin — APPROVED** [review]
This change is more on the aesthetical side of things to me. I am not opposed to but would like to get one more approval before merging it

**Review by jtraglia — APPROVED** [review]
I agree. This should be block hash. Thanks!

**Pull request merged**
